keadm/cmd/keadm/app/cmd/util/common_test.go

```diff
@@ -215,3 +215,165 @@ func TestComputeSHA512Checksum(t *testing.T) {
 		})
 	}
 }
+
+func TestKubeadmVersion(t *testing.T) {
+	type T struct {
+		name         string
+		input        string
+		output       string
+		outputError  bool
+		parsingError bool
+	}
+	cases := []T{
+		{
+			name:   "valid version with label and metadata",
+			input:  "v1.8.0-alpha.2.1231+afabd012389d53a",
+			output: "v1.8.0-alpha.2",
+		},
+		{
+			name:   "valid version with label and extra metadata",
+			input:  "v1.8.0-alpha.2.1231+afabd012389d53a.extra",
+			output: "v1.8.0-alpha.2",
+		},
+		{
+			name:   "valid patch version with label and extra metadata",
+			input:  "v1.11.3-beta.0.38+135cc4c1f47994",
+			output: "v1.11.2",
```

Comment:
Is this by design? 1.11.3-beta to 1.11.2

Reply:
Yes, as @fisherxu said, directly borrowed from the design of k8s.